### ISSUE TITLE
TS3633-CM1 Module

### ISSUE THREAD

**Issue by abencomo** [ashtuchkin/vive-diy-position-sensor#4]
Hi Alexander,

Will it possible to use the [TS3633-CM1 Module](https://www.triadsemi.com/product/ts3633-cm1/) instead of building your own [sensor board](https://cloud.githubusercontent.com/assets/627997/19845384/a4fce0e4-9ef3-11e6-95e4-6567ff374ee0.png)?

**Comment by ashtuchkin** [comment]
I haven't tried it yet, but it should.

**Comment by abencomo** [comment]
I ordered a few of the TS3633-CM1 Module so I'll let you know if it works.

I noticed that you use three(3) photodiodes optical sensors, but it seems that only one pin is connected from the Teensy to the TLV2462IP.  Are you fussing the values from the optical sensors?

BTW, I was able to build your project (i.e. the firmware) using eclipse as shown below.

<img width="1440" alt="screen shot 2016-12-06 at 11 23 40 am" src="https://cloud.githubusercontent.com/assets/7515273/20940306/1b76224c-bba7-11e6-853b-fe01adc71609.png">

**Comment by ashtuchkin** [comment]
Cool! I think I'll be able to experiment with TS3633-CM1 as well, but I definitely want to hear from you as well!

Yeah, those 3 photodiodes are connected in parallel to make it possible to cover the whole top hemisphere. It works fine, as the beam is almost as large as the gap between them and it looks like just a wider pulse.

I'm glad you compiled it in eclipse - did you need to do anything special? Should I add additional instructions?

**Comment by abencomo** [comment]
No, I didn't.  You did such a good job setting up the code that everything worked out of your repo.  BTW, are you in the bay area near SF?

**Comment by ashtuchkin** [comment]
That's great :) yeah I live in SF

**Comment by abencomo** [comment]
Great, I'm at Ames (Moffett Field) near you.  I should have the TS3633-CM1 by Thursday or Friday. Cheers!

**Comment by abencomo** [comment]
Hi Alex,

The TS3633-CM1 modules are in backorder so they haven't arrived yet.  In the meantime, I want to confirm a few things.

Are you using only one Vive Base Station in your video?  If that is the case, then did you switch it to mode "A"?

Is the Vive Base Station connected to your laptop via Bluetoorh?

And, do you need to have SteamVR running in your laptop?

Cheers!

**Comment by ashtuchkin** [comment]
Hey Al,

> Are you using only one Vive Base Station in your video? If that is the case, then did you switch it to mode "A"?

No, both are used. It's actually exactly the same setup as I use for regular Vive games. Modes B and C

> Is the Vive Base Station connected to your laptop via Bluetoorh?

I have a gaming desktop and they are probably connected via Bluetooth, yes.

> And, do you need to have SteamVR running in your laptop?

I have SteamVR running, but it's just to wake up the stations. AFAIK there's no realtime communication between the stations and desktop.

**Comment by abencomo** [comment]
Hum? That means you have a HMD in the loop.  Darn it.  It would have been great if only your IR receiving sensor with the Teensy and the base stations were the only components required.

**Comment by ashtuchkin** [comment]
Well, after waking stations up I can shut down desktop and hmd, so it's not
involved. Only base stations and sensor are needed.

**Comment by ashtuchkin** [comment]
You can also set up base stations to not go to sleep at all

**Comment by abencomo** [comment]
Keeping them from going to sleep is a good suggestion.  However, you still need the hmd to initiate the system before the quadcopter takes off, correct?

I also noticed that in your videos you're using a Mac laptop, were you able to run SteamVR on the Mac?

Thanks Alex for taking the time to clarify all these things.

**Comment by ashtuchkin** [comment]
​I'm not sure, but just connecting base stations to power seems to be
enough to initiate. The system's geometry is a different ​case - for now,
you need to calibrate the room using Vive's calibration and copy/paste the
matrices to the sensor. After that, only base stations are needed.

AFAIK SteamVR is not supported on Mac. I have a Windows gaming pc to play
vive.

--
Alexander Shtuchkin

**Comment by mpinner** [comment]
the TS3633-CM1 Module works great!

@ashtuchkin i made a blog post highlighting your work:
https://blog.crashspace.org/2016/12/position-aware-device/

thanks again!

**Comment by abencomo** [comment]
That is great. Thanks @mpinner for confirming that!  Now I cannot wait until mine arrived. Santa please, I have been good this year.

**Comment by ashtuchkin** [comment]
I just received TS3633-CM1 module and checked it's output with oscilloscope, looks great, should work with the code in this repo with minor adjustment due to inverted envelope:

![image](https://cloud.githubusercontent.com/assets/627997/21478457/79e8a09e-cb00-11e6-9cc6-4e075a85a7bd.png)
Channel 1: TS3633-CM1
Channel 2: Valve's own sensor (Alan Yates sent me a bunch)
Channel 3: My DIY sensor.

**Comment by abencomo** [comment]
Hi Alex,

Are you planning to update the repository with those minor adjustments to handle the inverted envelope?

**Comment by ashtuchkin** [comment]
Yep, as soon as I'm back from a vacation in mid jan. Open for PRs if you
need it faster, though.

--
Alexander Shtuchkin

**Comment by erwincoumans** [comment]
Very cool, can you explain how to connect the TS3633-CM1 to Teensy and what minor changes are needed?

**Comment by erwincoumans** [comment]
Great, I used Adrian Wong's change and connected the TS3633-CM1 to Teensy 3.2 and it works.
See image at the bottom here: https://github.com/erwincoumans/vive-diy-position-sensor#ts3633-cm1

How can we connect multiple TS3633-CM1 sensors?

**Comment by almostsquare** [comment]
Pull request for inverted envelope along with changes in anticipation of servicing 4+ TS3633-CM1 sensors is available at https://github.com/ashtuchkin/vive-diy-position-sensor/pull/10. Instructions on how to modify board wiring to run side-by-side is provided in the PR.

This is my first PR, so I welcome any feedback. I may not be structuring it correctly, and I'm happy to change things around as needed.
